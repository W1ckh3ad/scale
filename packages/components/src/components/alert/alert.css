.alert {
  box-sizing: border-box;
  background: #eee;
  color: #333;
  width: 100%;
  padding: 1rem;
  text-align: left;
  position: relative;
  display: flex;
  justify-content: space-between;
}
.alert__body {
  display: flex;
}
.alert__headline {
  font-size: var(--type-size-3, 1rem);
  color: white;
  margin: 0;
}
.alert__icon {
  display: flex;
  align-items: center;
  justify-content: center;
  background: none;
  height: 24px;
  width: 24px;
  margin: 0 0.5rem 0 0;
  border-radius: 4px;
}
.alert__close {
  height: 16px;
  opacity: 0.5;
  cursor: pointer;
}
.alert__close:hover {
  opacity: 1;
}
.alert--variant-primary {
  background: blue;
  color: #fff;
}
.alert--variant-secondary {
  background: #eee;
  color: #333;
}
.alert--variant-variant-warning {
  background: orange;
  color: #fff;
}
.alert--variant-danger {
  background: red;
  color: #fff;
}
.alert--variant-success {
  background: green;
  color: #fff;
}
.alert--variant-info {
  background: lightblue;
  color: #fff;
}
