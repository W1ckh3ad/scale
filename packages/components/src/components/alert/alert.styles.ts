import { JssStyle } from 'jss';

export const styles: JssStyle = {
  alert: {
    boxSizing: 'border-box',
    background: '#eee',
    color: '#333',
    width: '100%',
    padding: '1rem',
    textAlign: 'left',
    position: 'relative',
    display: 'flex',
    justifyContent: 'space-between',
  },
  alert__body: {
    display: 'flex',
  },
  alert__headline: {
    fontSize: '1rem',
    color: 'white',
    margin: 0,
  },
  alert__icon: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    background: 'none',
    height: '24px',
    width: '24px',
    margin: '0 0.5rem 0 0',
    borderRadius: '4px',
  },
  alert__close: {
    height: '16px',
    opacity: '0.5',
    cursor: 'pointer',
    '&hover': {
      opacity: 1,
    },
  },
  'alert--variant-primary': {
    background: 'blue',
    color: '#fff',
  },
  'alert--variant-secondary': {
    background: '#eee',
    color: '#333',
  },
  'alert--variant-variant-warning': {
    background: 'orange',
    color: '#fff',
  },
  'alert--variant-danger': {
    background: 'red',
    color: '#fff',
  },
  'alert--variant-success': {
    background: 'green',
    color: '#fff',
  },
  'alert--variant-info': {
    background: 'lightblue',
    color: '#fff',
  },
};
