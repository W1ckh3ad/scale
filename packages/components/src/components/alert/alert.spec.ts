import { newSpecPage } from '@stencil/core/testing';
import { Alert } from './alert';
import { styles } from './alert.styles';
import jss from 'jss';

describe('Alert', () => {
  let element;
  let stylesheet;
  beforeEach(async () => {
    element = new Alert();
    stylesheet = element.stylesheet = jss.createStyleSheet(styles as any);
    jest.useFakeTimers();
  });

  const components = [Alert];

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components,
      html: `<scale-alert>Notifications</scale-alert>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should match snapshot when opened', async () => {
    const page = await newSpecPage({
      components,
      html: `<scale-alert opened=true >Notifications</scale-alert>`,
    });
    expect(page.root.shadowRoot).toBeTruthy();
    expect(page.root).toMatchSnapshot();
  });

  it('should match snapshot with hasSlotClose', async () => {
    const page = await newSpecPage({
      components,
      html: `
      <scale-alert opened=true>
        Notifications
        <div slot="close">
          Close
        </div>
      </scale-alert>
      `,
    });
    expect(page.root.shadowRoot).toBeTruthy();
    expect(page.root).toMatchSnapshot();
  });

  it('should close the alert', () => {
    expect(element.opened).toBe(undefined);
    element.close();
    expect(element.opened).toBe(false);
  });

  it('should open the alert', () => {
    expect(element.opened).toBe(undefined);
    element.open();
    expect(element.opened).toBe(true);
  });

  it('should not open the alert/ should not render, if the alert is already opened', () => {
    element.opened = true;
    expect(element.root).toBeFalsy();
  });

  it('should closed the alert after default timeout', () => {
    element.timeout = true;
    element.onCloseAlertWithTimeout();

    expect(setTimeout).toHaveBeenCalledTimes(1);
    expect(setTimeout).toHaveBeenLastCalledWith(
      expect.any(Function),
      element.defaultTimeout
    );
  });

  it('should closed the alert after set timeout', () => {
    element.timeout = 500;
    element.onCloseAlertWithTimeout();

    expect(setTimeout).toHaveBeenCalledTimes(1);
    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 500);
  });

  it('should handle css classes', () => {
    element.customClass = 'custom';
    expect(element.getCssClassMap()).toContain('custom');

    element.size = 'small';
    stylesheet.addRule('alert--size-small', {});
    expect(element.getCssClassMap()).toContain(
      stylesheet.classes['alert--size-small']
    );

    element.variant = 'primary';
    stylesheet.addRule('alert--variant-primary', {});
    expect(element.getCssClassMap()).toContain(
      stylesheet.classes['alert--variant-primary']
    );
  });
});
