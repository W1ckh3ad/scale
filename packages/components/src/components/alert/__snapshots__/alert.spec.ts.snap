// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should match snapshot 1`] = `
<scale-alert>
  <mock:shadow-root></mock:shadow-root>
  Notifications
</scale-alert>
`;

exports[`Alert should match snapshot when opened 1`] = `
<scale-alert opened="">
  <mock:shadow-root>
    <style>
      .alert {
  color: #333;
  width: 100%;
  display: flex;
  padding: 1rem;
  position: relative;
  background: #eee;
  box-sizing: border-box;
  text-align: left;
  justify-content: space-between;
}
.alert__body {
  display: flex;
}
.alert__headline {
  color: white;
  margin: 0;
  font-size: 1rem;
}
.alert__icon {
  width: 24px;
  height: 24px;
  margin: 0 0.5rem 0 0;
  display: flex;
  background: none;
  align-items: center;
  border-radius: 4px;
  justify-content: center;
}
.alert__close {
  cursor: pointer;
  height: 16px;
  opacity: 0.5;
}
.alert__close:hover {
  opacity: 1;
}
.alert--variant-primary {
  color: #fff;
  background: blue;
}
.alert--variant-secondary {
  color: #333;
  background: #eee;
}
.alert--variant-variant-warning {
  color: #fff;
  background: orange;
}
.alert--variant-danger {
  color: #fff;
  background: red;
}
.alert--variant-success {
  color: #fff;
  background: green;
}
.alert--variant-info {
  color: #fff;
  background: lightblue;
}
    </style>
    <div class="alert">
      <div class="alert__body">
        <div class="alert__icon"></div>
        <div>
          <div class="alert__headline"></div>
          <slot></slot>
        </div>
      </div>
      <a class="alert__close">
        x
      </a>
    </div>
  </mock:shadow-root>
  Notifications
</scale-alert>
`;

exports[`Alert should match snapshot with hasSlotClose 1`] = `
<scale-alert opened="">
  <mock:shadow-root>
    <style>
      .alert {
  color: #333;
  width: 100%;
  display: flex;
  padding: 1rem;
  position: relative;
  background: #eee;
  box-sizing: border-box;
  text-align: left;
  justify-content: space-between;
}
.alert__body {
  display: flex;
}
.alert__headline {
  color: white;
  margin: 0;
  font-size: 1rem;
}
.alert__icon {
  width: 24px;
  height: 24px;
  margin: 0 0.5rem 0 0;
  display: flex;
  background: none;
  align-items: center;
  border-radius: 4px;
  justify-content: center;
}
.alert__close {
  cursor: pointer;
  height: 16px;
  opacity: 0.5;
}
.alert__close:hover {
  opacity: 1;
}
.alert--variant-primary {
  color: #fff;
  background: blue;
}
.alert--variant-secondary {
  color: #333;
  background: #eee;
}
.alert--variant-variant-warning {
  color: #fff;
  background: orange;
}
.alert--variant-danger {
  color: #fff;
  background: red;
}
.alert--variant-success {
  color: #fff;
  background: green;
}
.alert--variant-info {
  color: #fff;
  background: lightblue;
}
    </style>
    <div class="alert">
      <div class="alert__body">
        <div class="alert__icon"></div>
        <div>
          <div class="alert__headline"></div>
          <slot></slot>
        </div>
      </div>
      <a class="alert__close">
        <div>
          <slot name="close"></slot>
        </div>
      </a>
    </div>
  </mock:shadow-root>
  Notifications
  <div slot="close">
    Close
  </div>
</scale-alert>
`;
